src/shallow-equal.ts

```diff
@@ -1,3 +1,4 @@
+// @ts-nocheck
```

Comment:
is this prohibitively annoying to annotate with types?

alternatively this looks like an implementation we can use in place of this (which also has an `@types` package):
https://github.com/dashed/shallowequal/blob/master/index.js
https://www.npmjs.com/package/@types/shallowequal

alternative to the alternative: steal their type definitions lol